From 42a9ac7d36ac9ece7718718a38c37da72db8ea6a Mon Sep 17 00:00:00 2001
From: auto commit <unknown@unknown>
Date: Fri, 24 Oct 2008 12:23:12 -0700
Subject: [PATCH] cavium: hook MTD support into Makefile/Kconfig

Enable the build system to find and compile the Cavium MTD support.

Signed-off-by: Paul Gortmaker <Paul.Gortmaker@windriver.com>
---
 drivers/mtd/maps/Kconfig  |    8 ++++++++
 drivers/mtd/maps/Makefile |    1 +
 2 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/drivers/mtd/maps/Kconfig b/drivers/mtd/maps/Kconfig
index e626d89..2bd9ff2 100644
--- a/drivers/mtd/maps/Kconfig
+++ b/drivers/mtd/maps/Kconfig
@@ -571,5 +571,13 @@ config MTD_PLATRAM
 
 	  This selection automatically selects the map_ram driver.
 
+config MTD_CAVIUM_FLASH
+	tristate "Flash chip on Cavium CN38xx Boards"
+	depends on  CAVIUM_OCTEON_REFERENCE_BOARD && MTD
+	help
+           This provides a driver for the on-board boot flash.  By default the
+           flash is split into 1 partition which are accessed as separate
+           MTD devices.
+
 endmenu
 
diff --git a/drivers/mtd/maps/Makefile b/drivers/mtd/maps/Makefile
index 71603b7..bacb5a1 100644
--- a/drivers/mtd/maps/Makefile
+++ b/drivers/mtd/maps/Makefile
@@ -65,3 +65,4 @@ obj-$(CONFIG_MTD_PLATRAM)	+= plat-ram.o
 obj-$(CONFIG_MTD_OMAP_NOR)	+= omap_nor.o
 obj-$(CONFIG_MTD_INTEL_VR_NOR)	+= intel_vr_nor.o
 obj-$(CONFIG_MTD_BFIN_ASYNC)	+= bfin-async-flash.o
+obj-$(CONFIG_MTD_CAVIUM_FLASH)	+= cavium_flash.o
-- 
1.5.5.1

